src/app.js

```diff
@@ -1,37 +1,29 @@
 import { h } from "./h"
 
 export function app(props, container) {
+  var lock
+  var callbacks = []
   var root = (container = container || document.body).children[0]
   var node = toVNode(root, [].map)
-  var callbacks = []
-  var skipRender
-  var globalState
-  var globalActions
+  var globalState = {}
+  var globalActions = {}
 
-  repaint(flush(init(props, (globalState = {}), (globalActions = {}))))
+  repaint(init(props, globalState, globalActions, []))
 
   return globalActions
 
   function repaint() {
-    if (props.view && !skipRender) {
-      requestAnimationFrame(render, (skipRender = !skipRender))
+    if (props.view && !lock) {
+      setTimeout(render, (lock = !lock))
```

Comment:
What was the reasoning for changing `requestAnimationFrame` to `setTimeout`?

Reply:
From a discussion on slack @lukejacksonn found some drawbacks to `requestAnimationFrame` (not working on hidden tabs) plus the fact we needed to mock it for testing.

Reply:
setTimeout does not have that problems and still allow us to "debounce" re-renders after calling multiple actions in rapid-fire.